source/common/http/utility.cc

```diff
@@ -462,9 +462,19 @@ void Utility::appendVia(RequestOrResponseHeaderMap& headers, const std::string&
 
 void Utility::updateAuthority(RequestHeaderMap& headers, absl::string_view hostname,
                               const bool append_xfh) {
-  if (append_xfh && !headers.getHostValue().empty()) {
-    headers.appendForwardedHost(headers.getHostValue(), ",");
+  // only append to x-forwarded-host if the value was not the last value appended
```

Comment:
We try to keep comments using proper grammar
```suggestion
  // Only append to x-forwarded-host if the value was not the last value appended.
```